### ISSUE TITLE
Locale Format unit Uncaught TypeError

### ISSUE THREAD

**Issue by Jonaus** [grafana/grafana#11207]
Grafana v4.6.2 (commit: 8db5f08)
CentOS 7.4
InfluxDB 1.4.2

**Steps to reproduce**
1) Create new dashboard with graph panel
2) Apply following changes to panel:
2.1) **Metric:** Input any query (using default InfluxInternal works) using fill(null);
2.2) **Axes:** Use Locale Format for Y axis;
2.3) **Display:** In "Stacking & Null value" select "null" from "Null value" dropdown;
3) Hovering over empty data in graph shows no tooltip and produces `Uncaught TypeError: Cannot read property 'toLocaleString' of null` error on mouse movement.

tested using Chrome

**Comment by lookfirst** [comment]
Bug in 5.0.2 as well.

I love the [JS Errors Plugin](https://chrome.google.com/webstore/detail/javascript-errors-notifie/jafmfknfnkoekkdocjiaipcnmkklaajd) for detecting these sorts of errors in my own code.

Here is more details:

```
TypeError: Cannot read property 'toLocaleString' of null
 /public/build/app.c9bb2c091c2fc9c5fc35.js:1:24278 e.s.valueFormats.locale [as valueFormater]()
 /public/build/app.c9bb2c091c2fc9c5fc35.js:1:17430 e.formatValue()
 /public/build/0.62c2e3e7aa607c7f2d09.js:7:225687 n.show()
 /public/build/0.62c2e3e7aa607c7f2d09.js:7:224479 HTMLDivElement.()
 /public/build/vendor.84ef6926c3d7e7fc710d.js:87:34743 HTMLDivElement.dispatch()
 /public/build/vendor.84ef6926c3d7e7fc710d.js:87:32752 HTMLDivElement.g.handle()
 /public/build/vendor.84ef6926c3d7e7fc710d.js:87:55716 Object.trigger()
 /public/build/vendor.84ef6926c3d7e7fc710d.js:87:56237 HTMLDivElement.()
 /public/build/vendor.84ef6926c3d7e7fc710d.js:77:14819 Function.each()
 /public/build/vendor.84ef6926c3d7e7fc710d.js:77:13082 yt.fn.init.each()
```

**Comment by lookfirst** [comment]
Interesting... digging into this due to the `help wanted` tag...

This is the failing code line:

https://github.com/grafana/grafana/blob/a3388ef45f5aeb9a456a2cb8fa0c4871a4a59c63/public/app/core/utils/kbn.ts#L434

Then it begs the question on how to solve it. Check for null `value` or never pass a null `value` in. Also, what should the return value be if `value` is null?

Ah, but then there is this...

https://github.com/grafana/grafana/blob/11081010874c80aa4ed796335a73d86dd714d2cb/public/app/core/time_series2.ts#L314

Setting the value to null... and then passing in a null value. Doh!

Doing a bit of further reading as well...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

There is a section that says:

"Performance
When formatting large numbers of numbers, it is better to create a NumberFormat object and use the function provided by its NumberFormat.format property."

I wonder if this code can be further improved.

I'll let the experts decide, just wanted to convey the help. Cheers! 

;-)

**Comment by Bedasek** [comment]
Thanks @lookfirst for pointing out where the problem is.
Cuz I'm running **grafana-5.1.2** and from the repro steps posted above 2.3) is not necessary. It's enough to only meet criteria of 2.1) and 2.2) and having an actual null value in the graph.

Based on @lookfirst finding, I still wonder if this is really a "bug bug" or just an untreated scenario in the Grafana code...

**Comment by daniellee** [comment]
As is seems to be fixed by #14208 - closing this issue.

**Comment by lookfirst** [comment]
@daniellee See my comment above about performance implications. https://github.com/grafana/grafana/issues/11207#issuecomment-372594966
